packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -267,7 +267,10 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
                   // the dep list includes the main chunk, so only need to
                   // preload when there are actual other deps.
                   deps.size > 1
-                    ? `[${[...deps].map((d) => JSON.stringify(d)).join(',')}]`
+                    ? `[${[...deps]
+                        // replace leading slashes to ensure a relative path. See https://github.com/vitejs/vite/pull/3061
+                        .map((d) => JSON.stringify(d.replace(/^\//, './')))
```

Comment:
This doesn't seem to be the correct place to fix it. Check out how the array of deps is generated, the base is added to it:
https://github.com/vitejs/vite/blob/f3992984258ec47802f92f13fd99fc4e1747e1fd/packages/vite/src/node/plugins/importAnalysisBuild.ts#L245
Doesn't this fail if the base is not `/`? I think the paths in the deps array should be fixed in the `addDeps` function.

Reply:
Ups, I hadn't that in mind when I added my code. Thanks for pointing me at the right place to fix this. I added another commit and I also added a (quite artificial) test-case which plays around with the base-path.